### DISCUSSION TITLE
How to read response(NFC) from novopen echo plus

### DISCUSSION THREAD

**Discussion by RiteshChanchal1708** [NightscoutFoundation/xDrip#2673] [Ideas]
Hi, I am working on sample app to read dose history data from novopen. As per my understanding , we need to pass byte[] as parameter to isoDepInstance.transcieve().
Can someone help me here ? what bytearray we need to pass?

**Comment by C4s4r** [comment]
Have you found any solution on how to properly read and parse the NFC data?

**Comment by dgow** [comment]
there is an android app that can do this. may it helps https://github.com/lcacheux/nov-open-reader

https://f-droid.org/en/packages/net.cacheux.nvp.app/

**Reply by jamorham** [reply]
xDrip has had this support since 2022, look in the opennov package for a reference implementation. There certainly are many similarities with the structure and layout of the code in that repository you referenced. Its good to know that my contribution has been useful to people.
